PWGCF/Femto3D/Core/femto3dPairTask.h

```diff
@@ -0,0 +1,229 @@
+  FemtoPair() {}
+  FemtoPair(TrackType const& first, TrackType const& second)
+  {
+    _first = first;
+    _second = second;
+  }
+  FemtoPair(TrackType const& first, TrackType const& second, const bool& isidentical)
+  {
+    _first = first;
+    _second = second;
+    _isidentical = isidentical;
+  }
+
+  FemtoPair(const FemtoPair& obj)
+  {
+    SetFirstParticle(obj.GetFirstParticle());
+    SetSecondParticle(obj.GetSecondParticle());
+  }
+  FemtoPair(const FemtoPair* obj)
+  {
+    SetFirstParticle(obj->GetFirstParticle());
+    SetSecondParticle(obj->GetSecondParticle());
+  }
+  ~FemtoPair() {}
+  FemtoPair& operator=(const FemtoPair& obj)
+  {
+    if (this != &obj) {
+      SetFirstParticle(obj.GetFirstParticle());
+      SetSecondParticle(obj.GetSecondParticle());
+    }
+    return *this;
+  }
+
+  void SetPair(TrackType const& first, TrackType const& second)
+  {
+    _first = first;
+    _second = second;
+  }
+  void SetFirstParticle(TrackType const& first) { _first = first; }
+  void SetSecondParticle(TrackType const& second) { _second = second; }
+  void SetIdentical(const bool& isidentical) { _isidentical = isidentical; }
+  void SetMagField1(const float& magfield1) { _magfield1 = magfield1; }
+  void SetMagField2(const float& magfield2) { _magfield2 = magfield2; }
+  void SetPDG1(const int& PDG1) { _PDG1 = PDG1; }
+  void SetPDG2(const int& PDG2) { _PDG2 = PDG2; }
+  void ResetPair();
+  void ResetAll();
+
+  TrackType* GetFirstParticle() const { return _first; }
+  TrackType* GetSecondParticle() const { return _second; }
+  bool IsIdentical() { return _isidentical; }
+
+  bool IsClosePair(const float& deta = 0.01, const float& dphi = 0.01, const float& radius = 1.2);
+  float GetEtaDiff() const
+  {
+    if (_first != NULL && _second != NULL)
+      return _first->eta() - _second->eta();
+    else
+      return 1000;
+  }
+  float GetPhiStarDiff(const float& radius = 1.2) const
+  {
+    if (_first != NULL && _second != NULL)
+      return _first->phiStar(_magfield1, radius) - _second->phiStar(_magfield2, radius);
+    else
+      return 1000;
+  }
+  float GetKstar() const;
+  float GetKt() const;
+
+ private:
+  TrackType _first = NULL;
+  TrackType _second = NULL;
+  float _magfield1 = 0.0, _magfield2 = 0.0;
+  int _PDG1 = 0, _PDG2 = 0;
+  bool _isidentical = true;
+};
+
+template <typename TrackType>
+void FemtoPair<TrackType>::ResetPair()
+{
+  _first = NULL;
+  _second = NULL;
+}
+
+template <typename TrackType>
+void FemtoPair<TrackType>::ResetAll()
+{
+  _first = NULL;
+  _second = NULL;
+  _magfield1 = 0.0;
+  _magfield2 = 0.0;
+  _PDG1 = 0;
+  _PDG2 = 0;
+  _isidentical = true;
+}
+
+template <typename TrackType>
+bool FemtoPair<TrackType>::IsClosePair(const float& deta, const float& dphi, const float& radius)
+{
+  if (_first == NULL || _second == NULL)
+    return true;
+  if (!(_magfield1 * _magfield2))
+    return true;
+  if (abs(GetEtaDiff()) < deta || abs(GetPhiStarDiff(radius)) < dphi)
+    return true;
```

Comment:
Interesting! Shouldn't you require both?

Reply:
If you mean the values of magnetic field than it's the multiplication because when the "Pair" object is created the values are 0 by default and this is a simple insurance for both of them to be non-0 when the function is called (if at least one hasn't value hasn't been assigned and has the default value=0 then the multiplication=0).

Reply:
No, I meant having the separation in deta large but not in dphi is a bit inconsistent for a close pair

Reply:
Yes, you're right! Thanks a lot for spotting this. I just fixed.